pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -164,7 +168,7 @@ func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, optio
 				Name: aws.String(options.InstanceProfile),
 			},
 			SecurityGroupIds: aws.StringSlice(options.SecurityGroupsIDs),
-			UserData:         aws.String(options.UserData.Script()),
+			UserData:         aws.String(userData),
```

Comment:
Launch templates are currently cached just by the LT name, right? It would be difficult to hack on your userdata changes as you need to delete the LT every time or your change won't appear unless the LT has expired.

Reply:
LT name is a hash of the contents so one should get recreated.